meson.build

```diff
@@ -0,0 +1,129 @@
+project('mistserver', 'cpp', default_options: ['cpp_std=gnu++03'])
+
+ccpp = meson.get_compiler('cpp')
+
+incroot = include_directories('.')
+
+release = get_option('RELEASE')
+if release.contains('DEFAULT')
+    release = 'Generic_'+target_machine.cpu_family()
+endif
+release = release.strip()
+
+rv = run_command('git', 'describe', '--tags', check: false)
+version = rv.stdout().strip()
+if rv.returncode() != 0
+  version = 'Unknown'
+endif
+
+add_project_arguments(['-funsigned-char', '-D_LARGEFILE_SOURCE', '-D_FILE_OFFSET_BITS=64','-Wno-sign-compare', '-Wparentheses', '-Wno-non-virtual-dtor', '-Wno-non-virtual-dtor', '-Wno-strict-aliasing'], language: 'cpp')
```

Comment:
Wno-non-virtual-dtor is listed twice here.

Meson automatically passes `-D_FILE_OFFSET_BITS=64`, but not `-D_LARGEFILE_SOURCE`, since the former is safe by default but the latter depends on the project.

So at least you can get rid of one of them. :)